sstable/src/sstable_index.rs

```diff
@@ -16,10 +16,48 @@ impl SSTableIndex {
     }
 
     pub fn search_block(&self, key: &[u8]) -> Option<BlockAddr> {
+        self.search_block_id(key)
+            .and_then(|id| self.get_block_addr(id))
+    }
+
+    pub(crate) fn get_block_addr(&self, block_id: usize) -> Option<BlockAddr> {
         self.blocks
-            .iter()
-            .find(|block| &block.last_key_or_greater[..] >= key)
-            .map(|block| block.block_addr.clone())
+            .get(block_id)
+            .map(|block_meta| block_meta.block_addr.clone())
+    }
+
+    pub(crate) fn search_block_id(&self, key: &[u8]) -> Option<usize> {
```

Comment:
This function needs a doc... Especially if you were to handle multiple keys...

Reply:
> Especially if you were to handle multiple keys...

not sure what you mean here

Reply:
In the implementaton you handled the case where there were keys appearing several time in the Sstable.
This is good... But in that case, that method has an ambiguous behavior (is it returning the first block, the last block, or any block that might contain the key?) that should have been documented.